manifests/init.pp

```diff
@@ -12,4 +12,7 @@
       ensure => $package_ensure,
     } -> Rsync::Get<| |>
   }
+
+  create_resources(rsync::put, hiera_hash('rsync::puts', {}))
+  create_resources(rsync::get, hiera_hash('rsync::gets', {}))
```

Comment:
Instead of calling `hiera_hash()` in the manifest, it's better to add a parameter like `$puts` that accepts a hash and passes it to create_resources. I know it sounds odd, but there are lots of people who don't use hiera and hiera functions break their usage.